src/libexpr/primops.cc

```diff
@@ -368,8 +368,7 @@ void prim_exec(EvalState & state, const PosIdx pos, Value * * args, Value & v)
     auto output = runProgram(program, true, commandArgs);
     Expr * parsed;
     try {
-        auto base = state.positions[pos];
-        parsed = state.parseExprFromString(std::move(output), base.file);
+        parsed = state.parseExprFromString(std::move(output), "/");
```

Comment:
It doesn't really make sense to interpret `exec` relative to `pos` (since e.g. `exec args` would produce a different result than `{ f = args: exec args; }` and calling `f args` from another file) . Generally position info shouldn't be used for anything other than error messages or parse-time things like resolving relative paths.

Reply:
`./foo` is, but not `".nix"`. And I don't think that `if foo then import ./a else import ./b` is fundamentally different from `builtins.exec ["bash" "-c" "if [[ -n $FOO ]]; then echo 'import ./a' else echo 'import ./b'; fi"]`.

At any rate, it's not tremendously important. `builtins.exec` is this kind of thing that's explicitly dangerous and where you can reasonably expect minor breakages. It's just mildly annoying to have this kind of small and easy to avoid breakages

Reply:
It's a bug for a primop to depend on `pos` since that breaks referential transparency (e.g. inlining an expression that uses `exec` into another file could change its result). So I think it's better to fix that even if it's not directly relevant to this PR.